src/js/server/lib/password.js

```diff
@@ -0,0 +1,34 @@
+/*
+
+    A static function to consistently encode a password.  Used when storing passwords in CouchDB, and when logging in
+    using a username and password.
+
+    This package uses the `pbkdf2` functions provided by the `crypto` package built into node.  It has been tested to
```

Comment:
`pbkdf2` was used because it provided complete backward compatibility with the specific couch-native legacy accounts we had already.  This avoided asking everyone to reset their passwords.  We could certainly add support for other schemes later if desired.